src/renderer/hooks/usePaneNodeSearchMenu.tsx

```diff
@@ -0,0 +1,417 @@
+                                                onClick={() => {
+                                                    setSearchQuery('');
+                                                    onPaneContextMenuNodeClick(node, mousePosition);
+                                                }}
+                                            >
+                                                <IconFactory
+                                                    accentColor="gray.500"
+                                                    icon={node.icon}
+                                                />
+                                                <Text
+                                                    h="full"
+                                                    verticalAlign="middle"
+                                                >
+                                                    {node.name}
+                                                </Text>
+                                                (
+                                                {isFavorite && (
+                                                    <>
+                                                        <Spacer />
+                                                        <StarIcon
+                                                            aria-label="Favorites"
+                                                            boxSize={2.5}
+                                                            color="gray.500"
+                                                            overflow="hidden"
+                                                            stroke="gray.500"
+                                                            verticalAlign="middle"
+                                                        />
+                                                    </>
+                                                )}
+                                                )
+                                            </HStack>
+                                        );
+                                    })}
+                                </Box>
+                            );
+                        })
+                    ) : (
+                        <Center
+                            opacity="50%"
+                            w="full"
+                        >
+                            No compatible nodes found.
+                        </Center>
+                    )}
+                </Box>
+            </MenuList>
+        ),
+        [
+            connectingFrom,
+            connectingFromType,
+            byCategories,
+            onPaneContextMenuNodeClick,
+            searchQuery,
+            schemata.schemata,
+            matchingNodes,
+            mousePosition,
+        ]
+    );
+
+    useEffect(() => {
+        if (connectingFrom) {
+            const { nodeId, inOutId } = parseHandle(connectingFrom.handleId!);
```

Comment:
We can make the non-nullish assertion unnecessary like this.

```suggestion
        if (connectingFrom && connectingFrom.handleId) {
            const { nodeId, inOutId } = parseHandle(connectingFrom.handleId);
```